packages/react/src/utils/useCustomPages.tsx

```diff
@@ -48,16 +60,35 @@ type UseCustomPagesParams = {
   componentName: string;
 };
 
+type UseCustomPagesOptions = {
+  allowForAnyChildren: boolean;
+};
+
 type CustomPageWithIdType = UserProfilePageProps & { children?: React.ReactNode };
 
-const useCustomPages = ({
-  children,
-  LinkComponent,
-  PageComponent,
-  MenuItemsComponent,
-  reorderItemsLabels,
-  componentName,
-}: UseCustomPagesParams) => {
+export const useSanitizedChildren = (children: React.ReactNode) => {
```

Comment:
Can you add a JSDoc comment to explain what this function does? While reading it in the code that uses it it wasn't obvious to me what it does